ncm-network/src/main/perl/nmstate.pm

```diff
@@ -377,13 +393,13 @@ sub generate_nmstate_config
     #     next-hop-interface:
     #  and so on.
     my $routes = [];
+    push @$routes, @{$self->make_nm_route_absent($name)};
```

Comment:
Making an array_ref just to turn it back into a real array here seems odd.

I haven't seen what other similar make_nm_ subroutines do but from a complexity POV, could you return a hashref from make_nm_route_absent
```return { 'state' => 'absent', 'next-hop-interface' => $device };```

and then this becomes:

```
push @$routes, ($self->make_nm_route_absent($name));
```

Reply:
good shout, no idea why I didn't think of doing it like this. thanks :)